x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/adaptiveallocations/AdaptiveAllocationsScalerService.java

```diff
@@ -447,6 +450,12 @@ public boolean maybeStartAllocation(TrainedModelAssignment assignment) {
                 deploymentIdsWithInFlightScaleFromZeroRequests.add(assignment.getDeploymentId());
                 updateNumberOfAllocations(assignment.getDeploymentId(), 1, cleanUpListener);
             }
+
+            AdaptiveAllocationsScaler scaler = scalers.get(assignment.getDeploymentId());
+            if (scaler != null) {
+                scaler.resetTimeWithoutRequests();
```

Comment:
If starting an allocation takes too long (longer than the inference request timeout of 10secs), the inference request fails prematurely, and never reaches the allocation. This means that it isn't counted in the node stats, the "timeWithoutRequests" of the autoscaler keeps ticking, and the new allocation is stopped too soon.